src/claimtrie.cpp

```diff
@@ -2271,30 +2072,70 @@ expirationQueueType::iterator CClaimTrieCache::getSupportExpirationQueueCacheRow
 
 bool CClaimTrieCache::undoAddSupport(const std::string& name, const COutPoint& outPoint, int nHeight) const
 {
-    const std::string& newName = (base->shouldNormalize() ? normalizeClaimName(name) : name);
+    const std::string newName = base->normalizeClaimName(name);
     LogPrintf("%s: name: %s, txhash: %s, nOut: %d, nHeight: %d, nCurrentHeight: %d\n", __func__, newName, outPoint.hash.GetHex(), outPoint.n, nHeight, nCurrentHeight);
     int throwaway;
     return removeSupport(newName, outPoint, nHeight, throwaway, false);
 }
 
 bool CClaimTrieCache::spendSupport(const std::string& name, const COutPoint& outPoint, int nHeight, int& nValidAtHeight) const
 {
-    const std::string& newName = (base->shouldNormalize() ? normalizeClaimName(name) : name);
+    const std::string newName = base->normalizeClaimName(name);
     LogPrintf("%s: name: %s, txhash: %s, nOut: %d, nHeight: %d, nCurrentHeight: %d\n", __func__, newName, outPoint.hash.GetHex(), outPoint.n, nHeight, nCurrentHeight);
     return removeSupport(newName, outPoint, nHeight, nValidAtHeight, true);
 }
 
+bool CClaimTrieCache::normalizeAllNamesInTrieIfNecessary(insertUndoType& insertUndo, claimQueueRowType& removeUndo,
+        std::vector<std::pair<std::string, int> >& takeoverHeightUndo) const {
+    if (nCurrentHeight == Params().GetConsensus().nNormalizedNameForkHeight - 1) { // seems to get called twice per block
+
+        // run the one-time upgrade of all names that need to change
+        std::vector<namedNodeType> nodes = base->flattenTrie(); // TODO: replace all these calls with a real iterator (coded in #106)
+        for (std::vector<namedNodeType>::iterator it = nodes.begin(); it != nodes.end(); ++it) {
+            if (it->second.claims.empty()) continue;
+            const std::string normalized = base->normalizeClaimName(it->first, true);
+            if (normalized == it->first) continue;
+
+            LogPrintf("%s: Converting name on normalization fork upgrade: %s -> %s at %d\n", __func__, it->first.c_str(), normalized.c_str(), nCurrentHeight);
+
+            for(std::vector<CClaimValue>::iterator itClaim = it->second.claims.begin(); itClaim != it->second.claims.end(); ++itClaim) {
+
+                removeUndo.push_back(std::make_pair(it->first, *itClaim));
+                bool success = removeClaimFromTrie(it->first, itClaim->outPoint, *itClaim, false);
+                assert(success);
+
+                insertUndo.push_back(nameOutPointHeightType(normalized, itClaim->outPoint, itClaim->nValidAtHeight));
+                success = insertClaimIntoTrie(normalized, *itClaim, true);
+                assert(success);
+
+//                claimQueueEntryType entry(normalized, *itClaim);
```

Comment:
Why commented code?

Reply:
it was for a previous discussion and now removed